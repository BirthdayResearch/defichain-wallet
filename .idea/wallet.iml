<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <excludeFolder url="file://$MODULE_DIR$/.expo" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
      <excludeFolder url="file://$MODULE_DIR$/web-build" />
      <excludePattern pattern="*dist*" />
      <excludePattern pattern="*__snapshots__*" />
      <excludePattern pattern="*.nyc_output*" />
      <excludePattern pattern="*cypress-coverage*" />
      <excludePattern pattern="*jest-coverage*" />
      <excludePattern pattern="*reports*" />
    </content>
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>
