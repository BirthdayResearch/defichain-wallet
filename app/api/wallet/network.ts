import { MainNet, Network, RegTest, TestNet } from '@defichain/jellyfish-network'
import { EnvironmentNetwork } from '../../environment'

export function getJellyfishNetwork (network: EnvironmentNetwork): Network {
  switch (network) {
    case EnvironmentNetwork.MainNet:
      return MainNet
    case EnvironmentNetwork.TestNet:
      return TestNet
    case EnvironmentNetwork.LocalPlayground:
    case EnvironmentNetwork.RemotePlayground:
      return RegTest
  }
}
