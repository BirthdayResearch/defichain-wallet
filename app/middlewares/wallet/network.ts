import { MainNet, Network, RegTest, TestNet } from '@defichain/jellyfish-network'
import { EnvironmentNetwork } from '../../environment'
import { getNetwork } from '../storage'

export async function getNetworkOptions (): Promise<Network> {
  const network = await getNetwork()

  switch (network) {
    case EnvironmentNetwork.MainNet:
      return MainNet
    case EnvironmentNetwork.TestNet:
      return TestNet
    case EnvironmentNetwork.LocalPlayground:
    case EnvironmentNetwork.RemotePlayground:
      return RegTest
  }
}
