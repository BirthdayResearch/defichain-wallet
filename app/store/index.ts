/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { authentication } from './authentication'
import { block } from './block'
import { ocean } from './ocean'
import { transaction } from './transaction'
import { transactionQueue } from './transaction_queue'
import { wallet } from './wallet'

/**
 * RootState for DeFiChain Wallet App
 *
 * All state reducer in this store must be designed for global use and placed in this
 * directory as such. Reducer that are not meant to be global must not be part of
 * RootState.
 *
 * Non-global state should be managed independently within its own React Component.
 */
export function initializeStore () {
  return configureStore({
    reducer: {
      block: block.reducer,
      wallet: wallet.reducer,
      ocean: ocean.reducer,
      transactionQueue: transactionQueue.reducer,
      authentication: authentication.reducer,
      transaction: transaction.reducer
    },
    middleware: [
      ...getDefaultMiddleware({ serializableCheck: false })
    ]
  })
}

export type RootStore = ReturnType<typeof initializeStore>
export type RootState = ReturnType<RootStore['getState']>
