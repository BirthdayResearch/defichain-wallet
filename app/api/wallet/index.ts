import { JellyfishWallet, WalletHdNode, WalletHdNodeProvider } from '@defichain/jellyfish-wallet'
import { WhaleApiClient } from '@defichain/whale-api-client'
import { WhaleWalletAccount, WhaleWalletAccountProvider } from '@defichain/whale-api-wallet'
import { EnvironmentNetwork } from '../../environment'
import { getJellyfishNetwork } from './network'

/**
 * Whale JellyfishWallet connected to Whale APIs via the Ocean Infrastructure
 */
export type WhaleWallet = JellyfishWallet<WhaleWalletAccount, WalletHdNode>

export function initWhaleWallet (provider: WalletHdNodeProvider<WalletHdNode>, network: EnvironmentNetwork, client: WhaleApiClient): WhaleWallet {
  const accountProvider = new WhaleWalletAccountProvider(client, getJellyfishNetwork(network))
  return new JellyfishWallet(provider, accountProvider)
}

export * from './provider/mnemonic_encrypted'
export * from './provider/mnemonic_unprotected'
export * from './network'
export * from './passcode_attempt'
export * from './persistence'
