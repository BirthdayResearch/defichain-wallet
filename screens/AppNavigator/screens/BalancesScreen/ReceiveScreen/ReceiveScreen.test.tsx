import { configureStore } from "@reduxjs/toolkit";
import { fireEvent, render } from "@testing-library/react-native"
import * as React from 'react'
import { Provider } from "react-redux";
import Clipboard from 'expo-clipboard'
import { RootState } from "../../../../../store";
import { network } from "../../../../../store/network";
import { wallet, WalletStatus } from "../../../../../store/wallet";
import { ReceiveScreen } from "./ReceiveScreen";

jest.mock("expo-clipboard", () => ({
  setString: jest.fn()
}))

describe('receive page', () => {
  it('should match snapshot', async () => {
    const initialState: Partial<RootState> = {
      wallet: {
        address: 'bcrt1q6np0fh47ykhznjhrtfvduh73cgjg32yac8t07d',
        status: WalletStatus.LOADED_WALLET,
        utxoBalance: '77',
        tokens: []
      }
    };
    const store = configureStore({
      preloadedState: initialState,
      reducer: { wallet: wallet.reducer, network: network.reducer }
    })
    const component = (
      <Provider store={store}>
        <ReceiveScreen />
      </Provider>
    );
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('should trigger copy', async () => {
    const initialState: Partial<RootState> = {
      wallet: {
        address: 'bcrt1q6np0fh47ykhznjhrtfvduh73cgjg32yac8t07d',
        status: WalletStatus.LOADED_WALLET,
        utxoBalance: '77',
        tokens: []
      }
    };
    const store = configureStore({
      preloadedState: initialState,
      reducer: { wallet: wallet.reducer, network: network.reducer }
    })
    const component = (
      <Provider store={store}>
        <ReceiveScreen />
      </Provider>
    );
    const spy = jest.spyOn(Clipboard, 'setString')
    const rendered = render(component)
    const copyButton = await rendered.findByTestId('copy_button')
    fireEvent.press(copyButton)
    expect(spy).toHaveBeenCalled()
  })
})

