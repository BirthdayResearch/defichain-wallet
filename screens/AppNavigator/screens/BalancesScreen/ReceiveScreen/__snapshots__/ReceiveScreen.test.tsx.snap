// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receive page should match snapshot 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "500",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    Use this address to receive DFI or any DST
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 231, 235, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 231, 235, 1)",
        "borderRightColor": "rgba(229, 231, 235, 1)",
        "borderTopColor": "rgba(229, 231, 235, 1)",
        "display": "flex",
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
      testID="qr_code_container"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={200}
        bbWidth={200}
        focusable={false}
        height={200}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 200,
              "width": 200,
            },
          ]
        }
        vbHeight={200}
        vbWidth={200}
        width={200}
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  -65536,
                  1,
                  -16711681,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="grad"
              x1="0%"
              x2="100%"
              y1="0%"
              y2="100%"
            />
          </RNSVGDefs>
          <RNSVGGroup>
            <RNSVGRect
              fill={4294967295}
              height={200}
              propList={
                Array [
                  "fill",
                ]
              }
              width={200}
              x={-0}
              y={-0}
            />
          </RNSVGGroup>
          <RNSVGGroup>
            <RNSVGPath
              d="M0 3.0303030303030303 L42.42424242424242 3.0303030303030303 M60.60606060606061 3.0303030303030303 L72.72727272727272 3.0303030303030303 M78.78787878787878 3.0303030303030303 L90.9090909090909 3.0303030303030303 M103.03030303030303 3.0303030303030303 L109.0909090909091 3.0303030303030303 M115.15151515151516 3.0303030303030303 L127.27272727272727 3.0303030303030303 M133.33333333333334 3.0303030303030303 L151.5151515151515 3.0303030303030303 M157.57575757575756 3.0303030303030303 L200 3.0303030303030303 M0 9.09090909090909 L6.0606060606060606 9.09090909090909 M36.36363636363636 9.09090909090909 L42.42424242424242 9.09090909090909 M66.66666666666667 9.09090909090909 L72.72727272727272 9.09090909090909 M78.78787878787878 9.09090909090909 L109.0909090909091 9.09090909090909 M133.33333333333334 9.09090909090909 L139.3939393939394 9.09090909090909 M145.45454545454544 9.09090909090909 L151.5151515151515 9.09090909090909 M157.57575757575756 9.09090909090909 L163.63636363636363 9.09090909090909 M193.93939393939394 9.09090909090909 L200 9.09090909090909 M0 15.151515151515152 L6.0606060606060606 15.151515151515152 M12.121212121212121 15.151515151515152 L30.303030303030305 15.151515151515152 M36.36363636363636 15.151515151515152 L42.42424242424242 15.151515151515152 M48.484848484848484 15.151515151515152 L66.66666666666667 15.151515151515152 M78.78787878787878 15.151515151515152 L84.84848484848484 15.151515151515152 M96.96969696969697 15.151515151515152 L115.15151515151516 15.151515151515152 M157.57575757575756 15.151515151515152 L163.63636363636363 15.151515151515152 M169.6969696969697 15.151515151515152 L187.87878787878788 15.151515151515152 M193.93939393939394 15.151515151515152 L200 15.151515151515152 M0 21.21212121212121 L6.0606060606060606 21.21212121212121 M12.121212121212121 21.21212121212121 L30.303030303030305 21.21212121212121 M36.36363636363636 21.21212121212121 L42.42424242424242 21.21212121212121 M48.484848484848484 21.21212121212121 L66.66666666666667 21.21212121212121 M72.72727272727272 21.21212121212121 L78.78787878787878 21.21212121212121 M84.84848484848484 21.21212121212121 L103.03030303030303 21.21212121212121 M115.15151515151516 21.21212121212121 L139.3939393939394 21.21212121212121 M145.45454545454544 21.21212121212121 L151.5151515151515 21.21212121212121 M157.57575757575756 21.21212121212121 L163.63636363636363 21.21212121212121 M169.6969696969697 21.21212121212121 L187.87878787878788 21.21212121212121 M193.93939393939394 21.21212121212121 L200 21.21212121212121 M0 27.272727272727273 L6.0606060606060606 27.272727272727273 M12.121212121212121 27.272727272727273 L30.303030303030305 27.272727272727273 M36.36363636363636 27.272727272727273 L42.42424242424242 27.272727272727273 M48.484848484848484 27.272727272727273 L54.54545454545455 27.272727272727273 M60.60606060606061 27.272727272727273 L78.78787878787878 27.272727272727273 M90.9090909090909 27.272727272727273 L96.96969696969697 27.272727272727273 M109.0909090909091 27.272727272727273 L115.15151515151516 27.272727272727273 M127.27272727272727 27.272727272727273 L133.33333333333334 27.272727272727273 M145.45454545454544 27.272727272727273 L151.5151515151515 27.272727272727273 M157.57575757575756 27.272727272727273 L163.63636363636363 27.272727272727273 M169.6969696969697 27.272727272727273 L187.87878787878788 27.272727272727273 M193.93939393939394 27.272727272727273 L200 27.272727272727273 M0 33.333333333333336 L6.0606060606060606 33.333333333333336 M36.36363636363636 33.333333333333336 L42.42424242424242 33.333333333333336 M48.484848484848484 33.333333333333336 L84.84848484848484 33.333333333333336 M90.9090909090909 33.333333333333336 L115.15151515151516 33.333333333333336 M121.21212121212122 33.333333333333336 L127.27272727272727 33.333333333333336 M133.33333333333334 33.333333333333336 L139.3939393939394 33.333333333333336 M145.45454545454544 33.333333333333336 L151.5151515151515 33.333333333333336 M157.57575757575756 33.333333333333336 L163.63636363636363 33.333333333333336 M193.93939393939394 33.333333333333336 L200 33.333333333333336 M0 39.39393939393939 L42.42424242424242 39.39393939393939 M48.484848484848484 39.39393939393939 L54.54545454545455 39.39393939393939 M60.60606060606061 39.39393939393939 L66.66666666666667 39.39393939393939 M72.72727272727272 39.39393939393939 L78.78787878787878 39.39393939393939 M84.84848484848484 39.39393939393939 L90.9090909090909 39.39393939393939 M96.96969696969697 39.39393939393939 L103.03030303030303 39.39393939393939 M109.0909090909091 39.39393939393939 L115.15151515151516 39.39393939393939 M121.21212121212122 39.39393939393939 L127.27272727272727 39.39393939393939 M133.33333333333334 39.39393939393939 L139.3939393939394 39.39393939393939 M145.45454545454544 39.39393939393939 L151.5151515151515 39.39393939393939 M157.57575757575756 39.39393939393939 L200 39.39393939393939 M48.484848484848484 45.45454545454545 L66.66666666666667 45.45454545454545 M72.72727272727272 45.45454545454545 L78.78787878787878 45.45454545454545 M84.84848484848484 45.45454545454545 L96.96969696969697 45.45454545454545 M109.0909090909091 45.45454545454545 L115.15151515151516 45.45454545454545 M127.27272727272727 45.45454545454545 L139.3939393939394 45.45454545454545 M0 51.515151515151516 L6.0606060606060606 51.515151515151516 M12.121212121212121 51.515151515151516 L42.42424242424242 51.515151515151516 M66.66666666666667 51.515151515151516 L72.72727272727272 51.515151515151516 M78.78787878787878 51.515151515151516 L109.0909090909091 51.515151515151516 M133.33333333333334 51.515151515151516 L139.3939393939394 51.515151515151516 M157.57575757575756 51.515151515151516 L187.87878787878788 51.515151515151516 M0 57.57575757575758 L12.121212121212121 57.57575757575758 M24.242424242424242 57.57575757575758 L36.36363636363636 57.57575757575758 M54.54545454545455 57.57575757575758 L60.60606060606061 57.57575757575758 M72.72727272727272 57.57575757575758 L78.78787878787878 57.57575757575758 M115.15151515151516 57.57575757575758 L151.5151515151515 57.57575757575758 M157.57575757575756 57.57575757575758 L163.63636363636363 57.57575757575758 M181.8181818181818 57.57575757575758 L187.87878787878788 57.57575757575758 M193.93939393939394 57.57575757575758 L200 57.57575757575758 M12.121212121212121 63.63636363636364 L24.242424242424242 63.63636363636364 M36.36363636363636 63.63636363636364 L48.484848484848484 63.63636363636364 M60.60606060606061 63.63636363636364 L72.72727272727272 63.63636363636364 M90.9090909090909 63.63636363636364 L96.96969696969697 63.63636363636364 M109.0909090909091 63.63636363636364 L115.15151515151516 63.63636363636364 M127.27272727272727 63.63636363636364 L133.33333333333334 63.63636363636364 M145.45454545454544 63.63636363636364 L151.5151515151515 63.63636363636364 M163.63636363636363 63.63636363636364 L181.8181818181818 63.63636363636364 M187.87878787878788 63.63636363636364 L193.93939393939394 63.63636363636364 M6.0606060606060606 69.6969696969697 L12.121212121212121 69.6969696969697 M18.18181818181818 69.6969696969697 L24.242424242424242 69.6969696969697 M30.303030303030305 69.6969696969697 L36.36363636363636 69.6969696969697 M42.42424242424242 69.6969696969697 L54.54545454545455 69.6969696969697 M60.60606060606061 69.6969696969697 L66.66666666666667 69.6969696969697 M72.72727272727272 69.6969696969697 L78.78787878787878 69.6969696969697 M84.84848484848484 69.6969696969697 L96.96969696969697 69.6969696969697 M115.15151515151516 69.6969696969697 L121.21212121212122 69.6969696969697 M127.27272727272727 69.6969696969697 L139.3939393939394 69.6969696969697 M145.45454545454544 69.6969696969697 L163.63636363636363 69.6969696969697 M175.75757575757575 69.6969696969697 L187.87878787878788 69.6969696969697 M193.93939393939394 69.6969696969697 L200 69.6969696969697 M0 75.75757575757575 L6.0606060606060606 75.75757575757575 M12.121212121212121 75.75757575757575 L48.484848484848484 75.75757575757575 M54.54545454545455 75.75757575757575 L60.60606060606061 75.75757575757575 M66.66666666666667 75.75757575757575 L72.72727272727272 75.75757575757575 M109.0909090909091 75.75757575757575 L121.21212121212122 75.75757575757575 M127.27272727272727 75.75757575757575 L133.33333333333334 75.75757575757575 M139.3939393939394 75.75757575757575 L151.5151515151515 75.75757575757575 M163.63636363636363 75.75757575757575 L181.8181818181818 75.75757575757575 M193.93939393939394 75.75757575757575 L200 75.75757575757575 M12.121212121212121 81.81818181818181 L18.18181818181818 81.81818181818181 M30.303030303030305 81.81818181818181 L36.36363636363636 81.81818181818181 M42.42424242424242 81.81818181818181 L48.484848484848484 81.81818181818181 M60.60606060606061 81.81818181818181 L72.72727272727272 81.81818181818181 M78.78787878787878 81.81818181818181 L103.03030303030303 81.81818181818181 M121.21212121212122 81.81818181818181 L127.27272727272727 81.81818181818181 M133.33333333333334 81.81818181818181 L145.45454545454544 81.81818181818181 M151.5151515151515 81.81818181818181 L163.63636363636363 81.81818181818181 M175.75757575757575 81.81818181818181 L187.87878787878788 81.81818181818181 M193.93939393939394 81.81818181818181 L200 81.81818181818181 M24.242424242424242 87.87878787878788 L30.303030303030305 87.87878787878788 M36.36363636363636 87.87878787878788 L60.60606060606061 87.87878787878788 M66.66666666666667 87.87878787878788 L84.84848484848484 87.87878787878788 M115.15151515151516 87.87878787878788 L121.21212121212122 87.87878787878788 M127.27272727272727 87.87878787878788 L133.33333333333334 87.87878787878788 M151.5151515151515 87.87878787878788 L163.63636363636363 87.87878787878788 M175.75757575757575 87.87878787878788 L181.8181818181818 87.87878787878788 M187.87878787878788 87.87878787878788 L193.93939393939394 87.87878787878788 M18.18181818181818 93.93939393939394 L24.242424242424242 93.93939393939394 M30.303030303030305 93.93939393939394 L36.36363636363636 93.93939393939394 M54.54545454545455 93.93939393939394 L60.60606060606061 93.93939393939394 M66.66666666666667 93.93939393939394 L72.72727272727272 93.93939393939394 M78.78787878787878 93.93939393939394 L84.84848484848484 93.93939393939394 M96.96969696969697 93.93939393939394 L103.03030303030303 93.93939393939394 M109.0909090909091 93.93939393939394 L145.45454545454544 93.93939393939394 M157.57575757575756 93.93939393939394 L169.6969696969697 93.93939393939394 M181.8181818181818 93.93939393939394 L187.87878787878788 93.93939393939394 M193.93939393939394 93.93939393939394 L200 93.93939393939394 M0 100 L6.0606060606060606 100 M18.18181818181818 100 L24.242424242424242 100 M30.303030303030305 100 L42.42424242424242 100 M54.54545454545455 100 L66.66666666666667 100 M72.72727272727272 100 L78.78787878787878 100 M103.03030303030303 100 L109.0909090909091 100 M115.15151515151516 100 L121.21212121212122 100 M133.33333333333334 100 L139.3939393939394 100 M145.45454545454544 100 L157.57575757575756 100 M169.6969696969697 100 L181.8181818181818 100 M0 106.06060606060606 L6.0606060606060606 106.06060606060606 M12.121212121212121 106.06060606060606 L30.303030303030305 106.06060606060606 M60.60606060606061 106.06060606060606 L72.72727272727272 106.06060606060606 M84.84848484848484 106.06060606060606 L90.9090909090909 106.06060606060606 M96.96969696969697 106.06060606060606 L127.27272727272727 106.06060606060606 M139.3939393939394 106.06060606060606 L145.45454545454544 106.06060606060606 M151.5151515151515 106.06060606060606 L169.6969696969697 106.06060606060606 M181.8181818181818 106.06060606060606 L187.87878787878788 106.06060606060606 M193.93939393939394 106.06060606060606 L200 106.06060606060606 M0 112.12121212121212 L12.121212121212121 112.12121212121212 M18.18181818181818 112.12121212121212 L30.303030303030305 112.12121212121212 M36.36363636363636 112.12121212121212 L54.54545454545455 112.12121212121212 M60.60606060606061 112.12121212121212 L72.72727272727272 112.12121212121212 M84.84848484848484 112.12121212121212 L109.0909090909091 112.12121212121212 M133.33333333333334 112.12121212121212 L139.3939393939394 112.12121212121212 M145.45454545454544 112.12121212121212 L151.5151515151515 112.12121212121212 M157.57575757575756 112.12121212121212 L163.63636363636363 112.12121212121212 M187.87878787878788 112.12121212121212 L193.93939393939394 112.12121212121212 M0 118.18181818181819 L6.0606060606060606 118.18181818181819 M12.121212121212121 118.18181818181819 L24.242424242424242 118.18181818181819 M60.60606060606061 118.18181818181819 L72.72727272727272 118.18181818181819 M78.78787878787878 118.18181818181819 L127.27272727272727 118.18181818181819 M139.3939393939394 118.18181818181819 L145.45454545454544 118.18181818181819 M163.63636363636363 118.18181818181819 L169.6969696969697 118.18181818181819 M175.75757575757575 118.18181818181819 L200 118.18181818181819 M12.121212121212121 124.24242424242425 L42.42424242424242 124.24242424242425 M48.484848484848484 124.24242424242425 L66.66666666666667 124.24242424242425 M72.72727272727272 124.24242424242425 L84.84848484848484 124.24242424242425 M90.9090909090909 124.24242424242425 L103.03030303030303 124.24242424242425 M115.15151515151516 124.24242424242425 L139.3939393939394 124.24242424242425 M151.5151515151515 124.24242424242425 L157.57575757575756 124.24242424242425 M163.63636363636363 124.24242424242425 L181.8181818181818 124.24242424242425 M187.87878787878788 124.24242424242425 L193.93939393939394 124.24242424242425 M0 130.30303030303028 L6.0606060606060606 130.30303030303028 M18.18181818181818 130.30303030303028 L24.242424242424242 130.30303030303028 M30.303030303030305 130.30303030303028 L36.36363636363636 130.30303030303028 M42.42424242424242 130.30303030303028 L60.60606060606061 130.30303030303028 M90.9090909090909 130.30303030303028 L96.96969696969697 130.30303030303028 M109.0909090909091 130.30303030303028 L115.15151515151516 130.30303030303028 M139.3939393939394 130.30303030303028 L163.63636363636363 130.30303030303028 M181.8181818181818 130.30303030303028 L187.87878787878788 130.30303030303028 M193.93939393939394 130.30303030303028 L200 130.30303030303028 M0 136.36363636363637 L6.0606060606060606 136.36363636363637 M12.121212121212121 136.36363636363637 L24.242424242424242 136.36363636363637 M30.303030303030305 136.36363636363637 L48.484848484848484 136.36363636363637 M54.54545454545455 136.36363636363637 L60.60606060606061 136.36363636363637 M66.66666666666667 136.36363636363637 L78.78787878787878 136.36363636363637 M90.9090909090909 136.36363636363637 L109.0909090909091 136.36363636363637 M121.21212121212122 136.36363636363637 L127.27272727272727 136.36363636363637 M145.45454545454544 136.36363636363637 L151.5151515151515 136.36363636363637 M175.75757575757575 136.36363636363637 L181.8181818181818 136.36363636363637 M187.87878787878788 136.36363636363637 L193.93939393939394 136.36363636363637 M0 142.42424242424244 L6.0606060606060606 142.42424242424244 M30.303030303030305 142.42424242424244 L36.36363636363636 142.42424242424244 M48.484848484848484 142.42424242424244 L54.54545454545455 142.42424242424244 M66.66666666666667 142.42424242424244 L78.78787878787878 142.42424242424244 M84.84848484848484 142.42424242424244 L96.96969696969697 142.42424242424244 M109.0909090909091 142.42424242424244 L115.15151515151516 142.42424242424244 M127.27272727272727 142.42424242424244 L139.3939393939394 142.42424242424244 M145.45454545454544 142.42424242424244 L151.5151515151515 142.42424242424244 M163.63636363636363 142.42424242424244 L175.75757575757575 142.42424242424244 M181.8181818181818 142.42424242424244 L187.87878787878788 142.42424242424244 M0 148.48484848484847 L6.0606060606060606 148.48484848484847 M18.18181818181818 148.48484848484847 L30.303030303030305 148.48484848484847 M36.36363636363636 148.48484848484847 L54.54545454545455 148.48484848484847 M66.66666666666667 148.48484848484847 L84.84848484848484 148.48484848484847 M90.9090909090909 148.48484848484847 L109.0909090909091 148.48484848484847 M133.33333333333334 148.48484848484847 L139.3939393939394 148.48484848484847 M145.45454545454544 148.48484848484847 L181.8181818181818 148.48484848484847 M187.87878787878788 148.48484848484847 L200 148.48484848484847 M48.484848484848484 154.54545454545453 L66.66666666666667 154.54545454545453 M72.72727272727272 154.54545454545453 L78.78787878787878 154.54545454545453 M96.96969696969697 154.54545454545453 L103.03030303030303 154.54545454545453 M115.15151515151516 154.54545454545453 L133.33333333333334 154.54545454545453 M145.45454545454544 154.54545454545453 L151.5151515151515 154.54545454545453 M169.6969696969697 154.54545454545453 L187.87878787878788 154.54545454545453 M193.93939393939394 154.54545454545453 L200 154.54545454545453 M0 160.6060606060606 L42.42424242424242 160.6060606060606 M66.66666666666667 160.6060606060606 L96.96969696969697 160.6060606060606 M109.0909090909091 160.6060606060606 L115.15151515151516 160.6060606060606 M127.27272727272727 160.6060606060606 L133.33333333333334 160.6060606060606 M139.3939393939394 160.6060606060606 L151.5151515151515 160.6060606060606 M157.57575757575756 160.6060606060606 L163.63636363636363 160.6060606060606 M169.6969696969697 160.6060606060606 L175.75757575757575 160.6060606060606 M181.8181818181818 160.6060606060606 L193.93939393939394 160.6060606060606 M0 166.66666666666666 L6.0606060606060606 166.66666666666666 M36.36363636363636 166.66666666666666 L42.42424242424242 166.66666666666666 M48.484848484848484 166.66666666666666 L78.78787878787878 166.66666666666666 M90.9090909090909 166.66666666666666 L103.03030303030303 166.66666666666666 M115.15151515151516 166.66666666666666 L121.21212121212122 166.66666666666666 M127.27272727272727 166.66666666666666 L151.5151515151515 166.66666666666666 M169.6969696969697 166.66666666666666 L175.75757575757575 166.66666666666666 M181.8181818181818 166.66666666666666 L200 166.66666666666666 M0 172.72727272727272 L6.0606060606060606 172.72727272727272 M12.121212121212121 172.72727272727272 L30.303030303030305 172.72727272727272 M36.36363636363636 172.72727272727272 L42.42424242424242 172.72727272727272 M48.484848484848484 172.72727272727272 L60.60606060606061 172.72727272727272 M66.66666666666667 172.72727272727272 L96.96969696969697 172.72727272727272 M109.0909090909091 172.72727272727272 L121.21212121212122 172.72727272727272 M127.27272727272727 172.72727272727272 L133.33333333333334 172.72727272727272 M145.45454545454544 172.72727272727272 L181.8181818181818 172.72727272727272 M0 178.78787878787878 L6.0606060606060606 178.78787878787878 M12.121212121212121 178.78787878787878 L30.303030303030305 178.78787878787878 M36.36363636363636 178.78787878787878 L42.42424242424242 178.78787878787878 M48.484848484848484 178.78787878787878 L78.78787878787878 178.78787878787878 M84.84848484848484 178.78787878787878 L103.03030303030303 178.78787878787878 M121.21212121212122 178.78787878787878 L127.27272727272727 178.78787878787878 M133.33333333333334 178.78787878787878 L139.3939393939394 178.78787878787878 M145.45454545454544 178.78787878787878 L151.5151515151515 178.78787878787878 M169.6969696969697 178.78787878787878 L175.75757575757575 178.78787878787878 M181.8181818181818 178.78787878787878 L187.87878787878788 178.78787878787878 M193.93939393939394 178.78787878787878 L200 178.78787878787878 M0 184.84848484848484 L6.0606060606060606 184.84848484848484 M12.121212121212121 184.84848484848484 L30.303030303030305 184.84848484848484 M36.36363636363636 184.84848484848484 L42.42424242424242 184.84848484848484 M48.484848484848484 184.84848484848484 L54.54545454545455 184.84848484848484 M66.66666666666667 184.84848484848484 L78.78787878787878 184.84848484848484 M103.03030303030303 184.84848484848484 L109.0909090909091 184.84848484848484 M115.15151515151516 184.84848484848484 L121.21212121212122 184.84848484848484 M127.27272727272727 184.84848484848484 L133.33333333333334 184.84848484848484 M145.45454545454544 184.84848484848484 L163.63636363636363 184.84848484848484 M0 190.9090909090909 L6.0606060606060606 190.9090909090909 M36.36363636363636 190.9090909090909 L42.42424242424242 190.9090909090909 M72.72727272727272 190.9090909090909 L90.9090909090909 190.9090909090909 M96.96969696969697 190.9090909090909 L103.03030303030303 190.9090909090909 M109.0909090909091 190.9090909090909 L133.33333333333334 190.9090909090909 M139.3939393939394 190.9090909090909 L145.45454545454544 190.9090909090909 M151.5151515151515 190.9090909090909 L157.57575757575756 190.9090909090909 M175.75757575757575 190.9090909090909 L187.87878787878788 190.9090909090909 M0 196.96969696969697 L42.42424242424242 196.96969696969697 M48.484848484848484 196.96969696969697 L60.60606060606061 196.96969696969697 M66.66666666666667 196.96969696969697 L72.72727272727272 196.96969696969697 M78.78787878787878 196.96969696969697 L90.9090909090909 196.96969696969697 M103.03030303030303 196.96969696969697 L109.0909090909091 196.96969696969697 M115.15151515151516 196.96969696969697 L127.27272727272727 196.96969696969697 M133.33333333333334 196.96969696969697 L139.3939393939394 196.96969696969697 M145.45454545454544 196.96969696969697 L151.5151515151515 196.96969696969697 M163.63636363636363 196.96969696969697 L169.6969696969697 196.96969696969697 M175.75757575757575 196.96969696969697 L181.8181818181818 196.96969696969697 M187.87878787878788 196.96969696969697 L193.93939393939394 196.96969696969697 "
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                ]
              }
              stroke={4278190080}
              strokeLinecap={0}
              strokeWidth={6.0606060606060606}
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Object {
          "color": "rgba(107, 114, 128, 1)",
          "fontWeight": "500",
        }
      }
      testID="address_text"
    >
      bcrt1q6np0fh47ykhznjhrtfvduh73cgjg32yac8t07d
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
    testID="copy_button"
  >
    <Text />
    <Text
      style={
        Array [
          Object {
            "color": "#ff00af",
          },
          Object {
            "fontWeight": "500",
            "marginLeft": 8,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Copy
    </Text>
  </View>
</View>
`;
